freedomofpress/securedrop-workstation
# Uses grsecurity-patched kernels for all templates

Closes #156. 

Installs FPF-maintained custom kernels in the workstation TemplateVMs. Includes updates to config tests to confirm grsec-patched kernels are running.

### Testing

1. Run `make all` in `dom0` and confirm no errors.
2. Run `make test` in `dom0` and confirm no errors.
3. Confirm you can download and view a test submission. A plaintext submission should be displayed in `gedit` _in a DisposableVM_. 

I also confirmed that a JPG submission was displayed successfully, but it first tried to open Firefox, failed, then tried and succeeded with `/usr/bin/display`. The error message regarding Firefox is ugly, but we should resolve that behavior as part of #158, not here. 

Note also that we intend to remove these custom apt logic for the VM config via #157 in the near future. This PR still allows us to start using custom kernels and shake out the issues as they arise.
